park-dashboard/src/Components/AirQuality/AirQuality.jsx

```diff
@@ -1 +1,125 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "./AirQuality.css";
+import ReactSpeedometer from "react-d3-speedometer";
+
+function AirQuality(props) {
+  const [aqi, setAqi] = useState(0);
+  const [condition, setCondition] = useState();
+  const [description, setDescription] = useState();
+  const [color, setColor] = useState();
+
+  const parkCode = props.parkCode;
+
+  const getAQIConditionDescriptionAndColor = (aqi) => {
+    if (aqi >= 0 && aqi <= 50) {
+      return {
+        condition: "Good (0-50)",
+        description:
+          "Air quality is satisfactory, and there is little to no health risk.",
+        color: "#39b54930",
+      };
+    } else if (aqi > 50 && aqi <= 100) {
+      return {
+        condition: "Moderate (51-100)",
+        description:
+          "Air quality is acceptable, but certain pollutants may moderately affect sensitive individuals.",
+        color: "#ffff0050",
+      };
+    } else if (aqi > 100 && aqi <= 150) {
+      return {
+        condition: "Unhealthy for Sensitive Groups (101-150)",
+        description:
+          "Some individuals, particularly those with respiratory or heart conditions, may experience health issues due to the elevated pollution levels.",
+        color: "#f05a2250",
+      };
+    } else if (aqi > 150 && aqi <= 200) {
+      return {
+        condition: "Unhealthy (151-200)",
+        description:
+          "The general population may experience health effects, and sensitive individuals' conditions may worsen.",
+        color: "#ff000050",
+      };
+    } else if (aqi > 200 && aqi <= 300) {
+      return {
+        condition: "Very Unhealthy (201-300)",
+        description:
+          "The air quality is significantly compromised, and there is a higher risk of respiratory problems and other health issues for the general population.",
+        color: "#64017950",
+      };
+    } else {
+      return {
+        condition: "Hazardous (>300)",
+        description:
+          "Air quality is severely polluted, posing a significant risk to everyone's health, not just those with pre-existing conditions.",
+        color: "#9e010350",
+      };
+    }
+  };
+
+  const fetchAirQuality = (latitude, longitude) => {
+    axios
+      .get(
+        `https://api.api-ninjas.com/v1/airquality?lat=${latitude}&lon=${longitude}`,
+        {
+          headers: {
+            "X-Api-Key": "dOwjwEBWIUG+i0JD8MFslA==Fqr7PVBUnibAiqU3",
+          },
+        }
+      )
+      .then((res) => {
+        let aqi = res.data.overall_aqi;
+        setAqi(aqi);
+        const info = getAQIConditionDescriptionAndColor(aqi);
+        setColor(info.color);
+        setCondition(info.condition);
+        setDescription(info.description);
+      });
+  };
+  useEffect(() => {
+    const fetchParkData = async () => {
```

Comment:
Is it possible to use LoadParkData here since the same parks data is fetched?

Reply:
I tried making a call to LoadParkData within useEffect but it has run time errors while doing so. I think of keeping it simple and not reuse LoadParkData in this scenario. It could be because of making multiple API calls concurrently.

<img width="1502" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/114516950/953882f6-b1fc-4fa1-adb7-8af45ce4ebfd">